### PR TITLE
Request for app permissions simplified

### PR DESCRIPTION
**Pull request by GuybrushTh**
The app requests for permissions in `onCreate()` which causes that the permission request dialog appears at each start of the app (on some devices). This pull-request overcomes this issue by checking whether the permission are already granted and only requests them if necessary.

This fixes just the example **map-rendering-android**.

- `requestPermission()` only request permission if they are not granted yet
- introduced function `createMapFragmentView()`

### CODE REVIEW

**Comment by sschuberth** [comment]
Note to myself: This is similar to the work done by @saschpe for map-downloader in https://github.com/heremaps/here-android-sdk-examples/pull/26.

**Comment by GuybrushTh** [comment]
# Why does the build fail?
It seems there is an fault in `map-gesture:app` which I didn't even touch.

**Comment by sschuberth** [comment]
@GuybrushTh There's currently a limitation which requires PRs to originate from this very repo instead of a from fork (it's about Travis CI encrypted environment variables). Don't bother, we'll deal with this separately.

File: map-rendering/app/src/main/java/com/here/android/example/map/basic/MainActivity.java [hunk 1]
```diff
@@ -49,18 +50,48 @@ protected void onCreate(Bundle savedInstanceState) {
...
-        requiredSDKPermissions.add(Manifest.permission.ACCESS_NETWORK_STATE);
+        List<String> permissionList = Arrays.asList(
+                Manifest.permission.ACCESS_FINE_LOCATION,
+                Manifest.permission.WRITE_EXTERNAL_STORAGE,
+                Manifest.permission.INTERNET,
+                Manifest.permission.ACCESS_WIFI_STATE,
+                Manifest.permission.ACCESS_NETWORK_STATE);
+
+        // check list of permissions
+        Boolean isPermissionsGranted = true;
```
Comment (sschuberth) [inline map-rendering/app/src/main/java/com/here/android/example/map/basic/MainActivity.java]: Could you please make this a `boolean` (lower case)?

File: map-rendering/app/src/main/java/com/here/android/example/map/basic/MainActivity.java [hunk 2]
```diff
@@ -49,18 +50,48 @@ protected void onCreate(Bundle savedInstanceState) {
...
+        }
+
+        // return if all permissions already granted
+        if (isPermissionsGranted) {
+            createMapFragmentView();
+            return;
+        }
+
+        // request permissions
+        if (android.os.Build.VERSION.SDK_INT >= 23){
```
Comment (JohnLXiang) [inline map-rendering/app/src/main/java/com/here/android/example/map/basic/MainActivity.java]: Do we need to check the Android version here? The targetSDKVersion in the build.gradle has been set to 23

**Comment by JohnLXiang** [comment]
@GuybrushTh  Thanks for fixing this issue. Could we apply this change to all existing Android sample apps after this PR is approved?

File: map-rendering/app/src/main/java/com/here/android/example/map/basic/MainActivity.java (hunk 2 above)
Comment (GuybrushTh) [reply map-rendering/app/src/main/java/com/here/android/example/map/basic/MainActivity.java]: But the `minSdkVersion` is set to **16**.

Comment (sschuberth) [reply map-rendering/app/src/main/java/com/here/android/example/map/basic/MainActivity.java]: Yes, I think this is good as-is. We *could* replace the hard-coded `23` here with `Build.VERSION_CODES.M` like in PR #26, but personally I actually like seeing the number better than reading the code name.

**Comment by GuybrushTh** [comment]
@sschuberth txh for merging. 🌻

**Comment by sschuberth** [comment]
Thanks for contributing @GuybrushTh!

**Pull request merged**